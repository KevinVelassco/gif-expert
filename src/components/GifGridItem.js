import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {
    return (
        <div className="card  animate__animated animate__flip">
            <div className="card__imgContainer">
                <img src={url} alt={title} className="card__img" />
            </div>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired

}
