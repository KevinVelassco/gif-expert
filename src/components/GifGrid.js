import React, { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setimages] = useState([]);

    useEffect(() => {
        getGifs(category)
            .then(setimages)
    }, [category]);

    return (
        <>
            <h2>{category}</h2>
            <div className="panel">
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img} />
                    ))
                }
            </div>
        </>
    );
}