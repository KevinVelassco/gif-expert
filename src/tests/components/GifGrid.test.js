import React from 'react';
import { shallow } from "enzyme";
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test component <GifGrid />', () => {

    const category = 'anime';

    test('Debe mostrar el componente correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar items cuando se cargan las imÃ¡genes con useFetchGifs', () => {

        const gifs = [
            {
                id: 'sd312desad1',
                title: 'cualquier cosa',
                url: 'https://localhost/cualquier/cosa.jpg'
            }
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.loading').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    });

});