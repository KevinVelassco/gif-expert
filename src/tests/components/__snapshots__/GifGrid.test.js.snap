// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component <GifGrid /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <p
    className="category_content animate__animated animate__heartBeat"
  >
    <span
      className="category_title"
    >
      anime
    </span>
  </p>
  <p
    className="loading animate__animated animate__flash"
  >
    loading
  </p>
  <div
    className="panel"
  />
</Fragment>
`;

exports[`Test component <GifGrid /> Debe mostrar items cuando se cargan las im√°genes con useFetchGifs 1`] = `
<Fragment>
  <p
    className="category_content animate__animated animate__heartBeat"
  >
    <span
      className="category_title"
    >
      anime
    </span>
  </p>
  <div
    className="panel"
  >
    <GifGridItem
      id="sd312desad1"
      key="sd312desad1"
      title="cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
